client/context.go

```diff
@@ -18,6 +18,9 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+// ReformatTx allows chains to optionally reformat transactions before broadcasting
+type ReformatTxFn func(chainID string, key keyring.KeyType, tx TxBuilder) error
```

Comment:
This function signature is very non-intuitive. How does the reformatting occur? Where is the tx itself? Why is the TxBuilder called tx? How is someone who wants to reformat a tx meant to express their transformation?

A reformatter should take one version of a thing, and return the reformatted version.

```
type ReformatTxFn func(txBytes []byte) ([]byte, error)
```

If you intend to take an abstract Tx interface, then this isn't a Reformatter, it's just a Formatter, I suppose?